cpp/include/raft/sparse/linalg/detail/degree.cuh

```diff
@@ -54,29 +54,29 @@ RAFT_KERNEL coo_degree_kernel(const T* rows, int nnz, T* results)
  * @param results: output result array
  * @param stream: cuda stream to use
  */
-template <int TPB_X = 64, typename T = int>
-void coo_degree(const T* rows, int nnz, T* results, cudaStream_t stream)
+template <int TPB_X = 64, typename T = int, typename outT, typename nnz_t>
+void coo_degree(const T* rows, nnz_t nnz, outT* results, cudaStream_t stream)
 {
-  dim3 grid_rc(raft::ceildiv(nnz, TPB_X), 1, 1);
+  dim3 grid_rc(raft::ceildiv((nnz_t)nnz, (nnz_t)TPB_X), 1, 1);
   dim3 blk_rc(TPB_X, 1, 1);
 
   coo_degree_kernel<TPB_X><<<grid_rc, blk_rc, 0, stream>>>(rows, nnz, results);
   RAFT_CUDA_TRY(cudaGetLastError());
 }
 
-template <int TPB_X = 64, typename T>
-RAFT_KERNEL coo_degree_nz_kernel(const int* rows, const T* vals, int nnz, int* results)
+template <int TPB_X = 64, typename T, typename nnz_t>
+RAFT_KERNEL coo_degree_nz_kernel(const int* rows, const T* vals, nnz_t nnz, int* results)
 {
   int row = (blockIdx.x * TPB_X) + threadIdx.x;
   if (row < nnz && vals[row] != 0.0) { raft::myAtomicAdd(results + rows[row], 1); }
 }
 
-template <int TPB_X = 64, typename T>
+template <uint64_t TPB_X = 64, typename T, typename outT, typename nnz_t>
```

Comment:
This shuold be int always because it's an int type. Cast to nnz_t in place if you need this to match another type.